### PR TITLE
close aie context cleanup

### PR DESCRIPTION
**Pull request by SravanKumarAllu-xilinx**
#### Problem solved by the commit

shim is getting destroyed before Aie

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered

N/A

#### How problem was solved, alternative solutions (if any) and why they were rejected

resetting Aie in shim's destructor

#### Risks (if any) associated the changes in the commit

N/A

#### What has been tested and how, request additional testing if necessary

graph test cases with device / hwcontext

#### Documentation impact (if any)

N/A

### CODE REVIEW

File: src/runtime_src/core/edge/user/hwctx_object.cpp [hunk 1]
```diff
@@ -20,23 +20,35 @@ namespace zynqaie {
...
 
 #ifdef XRT_ENABLE_AIE
   std::shared_ptr<Aie>
   hwctx_object::get_aie_array_shared() const
   {
     return m_aie_array;
   }
+
+  aied*
+  hwctx_object::getAied()
```
Comment (stsoe) [inline src/runtime_src/core/edge/user/hwctx_object.cpp:33]: No camelCase

File: src/runtime_src/core/edge/user/hwctx_object.cpp [hunk 2]
```diff
@@ -20,23 +20,35 @@ namespace zynqaie {
...
+  {
+    return m_aied.get();
+  }
 #endif
 
+  void 
+  hwctx_object::initAie()
```
Comment (stsoe) [inline src/runtime_src/core/edge/user/hwctx_object.cpp:40]: ditto

File: src/runtime_src/core/edge/user/shim.cpp [hunk 3]
```diff
@@ -130,6 +130,11 @@ shim(unsigned index)
 shim::
 ~shim()
 {
+#ifdef XRT_ENABLE_AIE
+  if(aieArray)  // Aie cleanup should be done before shim destroyed
+    aieArray.reset();
```
Comment (stsoe) [inline src/runtime_src/core/edge/user/shim.cpp:135]:
```suggestion
  if (aie_array)  // Aie cleanup should be done before shim destroyed
    aie_array.reset();
```

**Comment by SravanKumarAllu-xilinx** [comment]
Due to merge conflicts, closing this PR , raising another PR with updated review comments

**Pull request closed without merging**